### PR TITLE
Ensure tests fail if any styles change

### PR DESCRIPTION
**Pull request by brettlangdon**
We have run into a few issues before of forgetting to run `grunt fmt` on the source code before accepting pull requests. This makes it a little difficult to go back and figure out what caused the changes.

This PR will add in format checking to `npm test`, where the build will fail if any files are reported as changing after running `jsfmt`.

Summary of changes:

* Added Grunt task `verify` to run a diff on the source code
    * Ideally this is run after running `grunt fmt`
    * It will display the diff before failing
* Added `fmt` command into `package.json`
    * We can now run `npm run fmt` instead of `grunt fmt`
* Updated `.travis.yml` to include more recent versions of `node.js` and `iojs`

This pull request is expected to fail the build. It passing it dependent on #173 being merged into here. I purposely did not stack the pull requests to help verify that this build fails in travis as expected.

### CODE REVIEW

**Comment by brettlangdon** [comment]
Here is an example failed build: https://travis-ci.org/rdio/jsfmt/jobs/87452016 which displays the diff before failing due to there being a diff.

**Comment by brettlangdon** [comment]
Updated here with master, which means these tests _should_ pass.

File: Gruntfile.js [hunk 1]
```diff
@@ -28,5 +29,6 @@ module.exports = function(grunt) {
   grunt.loadNpmTasks('grunt-exec');
   grunt.loadNpmTasks('grunt-mocha-test');
   grunt.registerTask('default', ['jshint', 'mochaTest']);
-  grunt.registerTask('fmt', ['exec']);
+  grunt.registerTask('fmt', ['exec:jsfmtLib', 'exec:jsfmtTests', 'exec:jsfmtGrunt', 'exec:jsfmtExamples']);
+  grunt.registerTask('verify', ['exec:verifyNoChanges']);
```
Comment (jimfleming) [inline Gruntfile.js]: Is `exec:verifyNoChanges` built-in somewhere? Could you add a comment? I couldn't find a reference in the source / Google.

**Comment by jimfleming** [comment]
This looks good. Tests were failing, I merged the PR and hopefully tests pass.

File: Gruntfile.js (hunk 1 above)
Comment (brettlangdon) [inline Gruntfile.js]: You mean the colon syntax? or the `verifyNoChanges` script?

**Comment by brettlangdon** [comment]
Don't merge yet. I can get them fixed here, looks like it is failing for some other reason as well.

**Comment by brettlangdon** [comment]
Ok, finally got this branch passing.

The changes represented now are:

* Update versions of node in travis
* Update travis to use new container infrastructure (tests are much faster now from the looks of it)
* Add in a `exec:verifyNoChanges` task to ensure `git diff` comes back clean
* Make sure to run `grunt fmt` on the source code and also new `grunt verify` task to ensure no changes effect styles.

File: Gruntfile.js (hunk 1 above)
Comment (jimfleming) [inline Gruntfile.js]: Ah, okay, now I see the exec task.

**Pull request merged**